### PR TITLE
[BD-6] Add tests with python 3.8

### PR DESCRIPTION
**Pull request by morenol**
## Description

Fix quality tests,
Upgrade python packages
Add tests for python 3.8

## Reviewers
- [ ] @ericfab179
- [ ] @awais786

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-4491](https://openedx.atlassian.net/browse/OSPR-4491) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by morenol** [comment]
Please review @awais786, I think that this is the kind of repo where we will need a final review from @jmbowman

File: requirements/constraints.txt [hunk 1]
```diff
@@ -8,9 +8,5 @@
 # pin when possible.  Writing an issue against the offending project and
 # linking to it here is good.
 
-# This packages is a backport which can only be installed on Python 2.7
-futures ; python_version == "2.7"
-
-# A dependency of pytest.  Pytest doesn't constrain it and 6.0.0 onward
-# only works with python 3
-more-itertools<6.0.0
+# Constraint from astroid 2.3.3
```
Comment (awais786) [inline requirements/constraints.txt:11]: why adding constraint here ?

Comment (morenol) [reply requirements/constraints.txt:11]: Make upgrade fails if that constraint is not present, I took that constraint fro  edx-platform

**Comment by morenol** [comment]
I think that were updated all the scripts in testeng to use python 3.5

The only missing thing is the upgrade job, there are two repos still using python 2.7, testeng-ci itself and edx-gomatic.

I guess that testeng-ci is not a problem since we can update that once this is merged, but what about edx-gomatic?

**Comment by morenol** [comment]
Done in https://github.com/edx/testeng-ci/pull/240

**Pull request closed without merging**